src/mon/MonmapMonitor.cc

```diff
@@ -129,7 +129,9 @@ void MonmapMonitor::create_pending()
   pending_map = *mon.monmap;
   pending_map.epoch++;
   pending_map.last_changed = ceph_clock_now();
-  dout(10) << __func__ << " monmap epoch " << pending_map.epoch << dendl;
+  pending_map.removed_ranks.clear();
```

Comment:
...yep, this was definitely supposed to happen here and not doing so was a big screw-up on my part. :/ Nice find!

If you look at the git history you'll see I said I expected this to always be of size 1, but I made it a container instead of an integer just in case I was missing some way for multiple removals to happen in a single map...really got that wrong. I should have made it an integer and asserted it wasn't set whenever a mon removal happened instead; that would have caught this and made it safe early. :/

Reply:
I see, I actually prefer we change the data type to int, However, because we are gearing towards applying the patch as an update for the customer's existing cluster I'm not sure if changing the data type will mess things up or not.